driver-async/src/main/com/mongodb/async/client/MongoCollectionImpl.java

```diff
@@ -300,7 +301,11 @@ public void bulkWrite(final List<? extends WriteModel<? extends TDocument>> requ
                 writeRequest = new DeleteRequest(toBsonDocument(deleteManyModel.getFilter())).multi(true)
                         .collation(deleteManyModel.getOptions().getCollation());
             } else {
-                throw new UnsupportedOperationException(format("WriteModel of type %s is not supported", writeModel.getClass()));
+                if (writeModel == null) {
```

Comment:
No real need for an extra level of conditional here.  I actually think it would be clearer to leave this else as is, and move the null check to the very top before all the instance checks.